### PR TITLE
feat: Update Braze to 4.2.1

### PR DESCRIPTION
**Pull request by rmi22186**
## Summary
- Reopening this PR made by @spvismaya.  There were some conflicts after we merged a few PRs in after she initially PR'd and so I thought it was easier for us to redo the PR.

## Testing Plan
- Updated the test suite as needed and they all passed

## Reference Issue
- Closes https://go.mparticle.com/work/SQDSDKS-4297

### CODE REVIEW

**Comment by rmi22186** [comment]
@alexs-mparticle 
I had to update the rollup version because Braze started to do tree shaking in v4, which isn't compatible with rollup v1 for our requirements since we are requiring the full SDK. I was getting the following error - https://stackoverflow.com/questions/65690682/why-does-rollup-complain-about-code-splitting-when-i-am-not-code-splitting

The solutio is to add `inlineDynamicImports: true` to the rollup config, but this requires version 2 of rollup.

File: src/BrazeKit-dev.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* eslint-disable no-undef */
-window.appboy = require('@braze/web-sdk');
+window.braze = require('@braze/web-sdk');
```
Comment (alexs-mparticle) [inline src/BrazeKit-dev.js:2]: Shouldn't you change line 17 from `Appboy` to `Braze`?

```
var name = 'Appboy',
    moduleId = 28,
    version = '3.0.4',
    MessageType = {
        PageView: 3,
        PageEvent: 4,
        Commerce: 16,
    };
```

Comment (alexs-mparticle) [reply src/BrazeKit-dev.js:2]: OK, please provide a comment with a link to the explanation

File: src/BrazeKit-dev.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ window.appboy = require('@braze/web-sdk');
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
+// This should remain Appboy and not Braze until the core SDK is able to parse the moduleID and not the name (go.mparticle.com/SQDSDKS-4655)
```
Comment (alexs-mparticle) [inline src/BrazeKit-dev.js:17]:
```suggestion
// This should remain Appboy and not Braze until the core SDK is able to parse the moduleID and not the name (go.mparticle.com/work/SQDSDKS-4655)
```

**Review by alexs-mparticle — APPROVED** [review]

**Pull request merged**